### ISSUE TITLE
Amazon Linux easy_install incompatible with 2.14.1

### ISSUE THREAD

**Issue by jfwarner** [kennethreitz/requests#4013]
I've been following #4006 #4012 and I'm aware the guidance is to update to a newer pip, and that easy_install is not supported. I just wanted to make the case that not all of us have a choice on python package manager.

Take my case - I'm using Amazon Linux bootstrapped by Amazon CloudFormation. We give the CloudFormation initializer code (cfn-init for short) a list of python packages to install, and it installs them in any order it wants, using any mechanism it wants. Studying the logs, it looks like under the hood it's using easy_install.

I'm not even trying to install requests. I'm getting a floating version of it transitively through trying to install docker-py. So I see:

`[ERROR] easy_install failed. Output: Searching for docker-py==1.7.2`
...
`Searching for requests>=2.5.2`
...
`Best match: requests 2.14.1`
...
```
Running requests-2.14.1/setup.py -q bdist_egg --dist-dir /tmp/easy_install-wfkGhb/requests-2.14.1/egg-dist-tmp-H3aUMy
error: Setup script exited with error in requests setup command: Invalid environment marker: sys_platform == "win32" and python_version<"3.3"
```

Version information:
Amazon Linux AMI release 2016.09
Python 2.7.12
```
easy_install --version
setuptools 12.2
```

So at the moment I have a cumbersome workaround where I run cfn-init in two phases, first where I expressly install requests 2.13.0, then next where I install docker-py and it's happy with the requests already installed. But I just wanted to raise awareness that there is a category of people who are seeing this error through transitive use of requests (I didn't even know we were using it) and for which upgrading or changing package managers is non-trivial.

**Comment by Lukasa** [comment]
@jfwarner Thanks for this report.

Is it not possible to [upgrade the version of setuptools before you run cloudformation](https://gist.github.com/kixorz/10194688)? That will likely resolve your issue.

**Comment by jfwarner** [comment]
Unfortunately setuptools 12.2 is the latest and greatest available from Amazon Linux's yum repo.

```
yum update python27-setuptools
Loaded plugins: priorities, update-motd, upgrade-helper
amzn-main
amzn-updates
No packages marked for update
```

```
yum info python27-setuptools
Loaded plugins: priorities, update-motd, upgrade-helper
Installed Packages
Name        : python27-setuptools
Arch        : noarch
Version     : 12.2
Release     : 1.32.amzn1
Size        : 1.9 M
Repo        : installed
```

**Comment by Lukasa** [comment]
Gah, that's pretty frustrating. This seems to be down to setuptools 12 not knowing how to use range operators in markers. We can probably fix it though. See #4014.

**Comment by jfwarner** [comment]
Thanks - I know it's our own fault for not locking down all our transitive dependencies, but this is causing us a production outage on the ability to boot new servers.

**Comment by jfwarner** [comment]
Confirmed that the issue has been resolved in 2.14.2 - thanks!

**Comment by Lukasa** [comment]
No problem, glad we could help!
